### ISSUE TITLE
Incorrect result with ROLLUP and HAVING, with optimizer=off and on PostgreSQL

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#3507]
On PostgreSQL, or on GPDB with optimizer=off:
```
postgres=# SELECT COUNT(DISTINCT cn) as cn_r, f, g FROM (SELECT cn, vn + 2 AS f, 2 AS g FROM sale) sale_view GROUP BY ROLLUP(f,g) HAVING (g > 1);
 cn_r | f  | g 
------+----+---
    1 | 12 | 2
    1 | 12 |  
    1 | 22 | 2
    1 | 22 |  
    2 | 32 | 2
    2 | 32 |  
    3 | 42 | 2
    3 | 42 |  
    2 | 52 | 2
    2 | 52 |  
    4 |    |  
(11 rows)
```

I believe that's incorrect. The HAVING clause should've filtered out the rows where g is NULL. ORCA produces the expected result with only 5 rows.

**Comment by hlinnaka** [comment]
Reported to pgsql-bugs, with a patch: https://www.postgresql.org/message-id/7dbdcf5c-b5a6-ef89-4958-da212fe10176%40iki.fi

**Comment by yydzero** [comment]
Is this still an issue? Looks there are a few fixes related with this.

**Comment by hlinnaka** [comment]
> Is this still an issue? Looks there are a few fixes related with this.

The bug still exists.

This was fixed in upstream commit 90947674fc984f5639e3b1bf013435a023aa713b. It was only backpatched down to 9.5, however, because GROUPING SETS were introduced in that version. I haven't checked if the upstream fix would be applicable to the GPDB implementation of GROUPING SETS.

**Comment by adam8157** [comment]
The case is in the master branch now (thanks to the merge project), and it's OK. Close this.
